internal/engine/engine.go

```diff
@@ -274,6 +277,31 @@ func (engine *Engine) evaluate(ctx context.Context, input *enginev1.CheckInput,
 		Actions:    make(map[string]*enginev1.CheckOutput_ActionEffect, len(input.Actions)),
 	}
 
+	err = engine.schemaMgr.Validate(ctx, input)
+	if err != nil && !schema.IsValidationErrorList(err) {
+		return nil, err
+	}
+
+	var validationErrorList *schema.ValidationErrorList
+	ok := errors.As(err, &validationErrorList)
+	if !ok {
+		validationErrorList = nil
+	}
+
+	// If there are errors present in schema validation, return EFFECT_DENY for all actions
+	if validationErrorList != nil && len(validationErrorList.Errors) > 0 {
+		output.ValidationErrors = validationErrorList.SchemaErrors()
+
+		for _, action := range input.Actions {
+			output.Actions[action] = &enginev1.CheckOutput_ActionEffect{
+				Effect: effectv1.Effect_EFFECT_DENY,
+				Policy: noPolicyMatch,
```

Comment:
Let's change this to a value like "skipped due to invalid input" or something like that.